salt/preflight/mandatory.sls

```diff
@@ -0,0 +1,45 @@
+{% import_yaml 'preflight/defaults.yaml' as settings %}
+
+install mandatory packages:
+  pkg.installed:
+    - pkgs: {{ settings.mandatory.packages }}
+
+{%- for directory in settings.mandatory.available_pathdirs %}
+make sure {{ directory }} does not exist:
+  file.absent:
```

Comment:
This would wipe the `etcd` database, for example, when executed. That's not what we want :smiley: because this state should also be evaluated upon upgrade.

Maybe we should split things up in
- a set of assertions which should be true when a node joins the cluster, or becomes part of the control plane, e.g. there should be no `etcd` database
- a set of configuration checks we can non-destructively apply when a node is set up, e.g. `sysctl` settings

We could at some point code up a state which brings the node to `reset` state, e.g. by removing the `etcd` database, but that's definitely optional because very dangerous, and should never ever be invoked automatically.